paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheet.kt

```diff
@@ -38,6 +37,25 @@ class CustomerSheet internal constructor(
         )
     }
 
+    private fun onCustomerSheetResult(result: InternalCustomerSheetResult?) {
+        requireNotNull(result)
+        val customerSheetResult = when (result) {
```

Comment:
I typically prefer to do method dispatch over when statements like this.

Do you think it's worth adding a `toPublicResult` on `InternalCustomerSheetResult`?

Then we can just map these all in each subtype. The down side is it's typically a bit more code. The upside is it's really clear that changing something in the internal will need to map to external. Keeping all that logic mapping close.